lib/shared/sdk/usbboot/index.js

```diff
@@ -142,6 +142,41 @@ const USB_ID_RADIX = 16
 const USB_ID_LENGTH = 4
 
 /**
+ * @summary The cache of blobs
+ * @type {Object}
+ * @constant
+ */
+const BLOBS_CACHE = {}
+
+/**
+ * @summary Query a blob from the internal cache
+ * @function
+ * @private
+ *
+ * @param {String} name - blob name
+ * @param {Function} fallback - fallback function
+ * @fulfil {Buffer} - blob
+ * @returns {Promise}
+ *
+ * @example
+ * const Bluebird = require('bluebird')
+ * const fs = Bluebird.promisifyAll(require('fs'))
+ *
+ * const blob = queryBlobFromCache('start.elf', (name) => {
+ *   return fs.readFileAsync(path.join('./blobs', name))
+ * })
+ */
+const queryBlobFromCache = (name, fallback) => {
+  if (BLOBS_CACHE[name]) {
+    return Bluebird.resolve(BLOBS_CACHE[name])
+  }
+
+  return fallback(name).tap((buffer) => {
+    BLOBS_CACHE[name] = buffer
+  })
+}
```

Comment:
Doesn't really matter, but when I've written code like this before I tend to do something like the following pseudocode:
```
function readCachedFile(filename) {
    if (filename not in cache) {
        cache[filename] = readFile(filename);
    }
    return cache[filename];
}
```
but perhaps that wouldn't work with the Promise-based stuff you're doing here, which I still struggle with sometimes ;-)

Reply:
Yeah, promises make that a bit hard.